core/src/main/java/fr/sncf/osrd/stdcm/graph/STDCMSimulations.java

```diff
@@ -65,26 +66,43 @@ public static Envelope simulateRoute(
             RollingStock rollingStock,
             RollingStock.Comfort comfort,
             double timeStep,
-            double[] stops,
+            double stopPosition,
             String tag
     ) {
+        if (stopPosition == 0)
+            return makeSinglePointEnvelope(0);
+        var context = makeSimContext(List.of(route), start, rollingStock, comfort, timeStep);
+        double[] stops = new double[]{};
+        double length = context.path.getLength();
+        if (!Double.isNaN(stopPosition)) {
```

Comment:
This seems dangerous. I'd prefer an Optional/OptionalDouble (or just a boxed Double) instead of relying on NaN to convey an absent value.